bot/src/infrastructure/repositories/MapActivityRepository.ts

```
@@ -234,131 +403,197 @@
+
+        // Format string for grouping
+        const getTimeKey = (date: Date): string => {
+          switch (groupBy) {
+            case 'hour':
+              return `${date.getFullYear()}-${date.getMonth()}-${date.getDate()}-${date.getHours()}`;
+            case 'week': {
+              const d = new Date(date);
+              d.setDate(d.getDate() - d.getDay()); // Start of week (Sunday)
+              return `${d.getFullYear()}-${d.getMonth()}-${d.getDate()}`;
+            }
+            case 'day':
```

Biome (1.9.4)

[error] Useless case clause.

because the default clause is present:

Unsafe fix: Remove the useless case.


(lint/complexity/noUselessSwitchCase)